src/sentry/monitors/serializers.py

```diff
@@ -11,7 +11,58 @@
 from sentry.monitors.validators import IntervalNames
 
 from ..models import Environment
-from .models import Monitor, MonitorCheckIn, MonitorEnvironment, MonitorStatus
+from .models import (
+    Monitor,
+    MonitorCheckIn,
+    MonitorEnvBrokenDetection,
+    MonitorEnvironment,
+    MonitorIncident,
+    MonitorStatus,
+)
+
+
+class MonitorEnvBrokenDetectionSerializerResponse(TypedDict):
+    userNotifiedTimestamp: datetime
+    environmentMutedTimestamp: datetime
+
+
+@register(MonitorEnvBrokenDetection)
+class MonitorEnvBrokenDetectionSerializer(Serializer):
+    def serialize(self, obj, attrs, user, **kwargs) -> MonitorEnvBrokenDetectionSerializerResponse:
+        return {
+            "userNotifiedTimestamp": obj.user_notified_timestamp,
+            "environmentMutedTimestamp": obj.env_muted_timestamp,
+        }
+
+
+class MonitorIncidentSerializerResponse(TypedDict):
+    startingTimestamp: datetime
+    resolvingTimestamp: datetime
+    brokenNotice: MonitorEnvBrokenDetectionSerializerResponse | None
+
+
+@register(MonitorIncident)
+class MonitorIncidentSerializer(Serializer):
+    def get_attrs(
+        self, item_list: Sequence[Any], user: Any, **kwargs: Any
+    ) -> MutableMapping[Any, Any]:
+        broken_detections = {
+            broken_detection.monitor_incident_id: broken_detection
+            for broken_detection in MonitorEnvBrokenDetection.objects.filter(
+                monitor_incident__in=item_list
+            )
+        }
+        return {
+            incident: {"broken_detection": (serialize(broken_detections.get(incident.id), user))}
```

Comment:
Don't we want to serialize the list as well as part of the get_attrs, not inside of this for loop. Otherwise, we may introduce another n+1 if we add a query into the broken detections serializer?

cc @wedamija

Reply:
Right, yeah that can be a good idea too. Up to you both if you want to do it proactively, since it doesn't cause a problem today

Reply:
Easy enough to do with
```
    broken_detections = list(MonitorEnvBrokenDetection.objects.filter(
            monitor_incident__in=item_list
    ))
    serialized_broken_detections = {detection.id: serialized for serialized, detection in zip(serialize(broken_detections, user), broken_detections)
 ```

Reply:
lol I just wrote the same thing down here https://github.com/getsentry/sentry/pull/66989#discussion_r1525454789

Reply:
oooh lol yes should definitely serialize the list here as well, ty for the help yall 😄